packages/client/src/runtime/core/model/aggregates/aggregate.ts

```diff
@@ -0,0 +1,82 @@
+import type { Client } from '../../../getPrismaClient'
+import type { ModelAction } from '../applyModel'
+import { aggregateMap } from './utils/aggregateMap'
+
+/**
+ * Transforms the `userArgs` for the `.aggregate` shorthand. It is an API sugar
+ * for not having to do things like: `{select: {_avg: {select: {age: true}}}}`.
+ * The goal here is to desugar it into something that is understood by the QE.
+ * @param userArgs to transform
+ * @returns
+ */
+export function desugarUserArgs(userArgs: object) {
+  const _userArgs = desugarCountInUserArgs(userArgs)
+  const userArgsEntries = Object.entries(_userArgs)
+
+  return userArgsEntries.reduce(
```

Comment:
Since the accumulator here is mutated rather than constructed immutably, it could've been just a `forEach` or a `for` loop with one or two lines of code less, but no strong opinion about this, `reduce` is fine too.